packages/@aws-cdk/aws-ecs/lib/ec2/ec2-service.ts

```diff
@@ -154,8 +154,13 @@ export class Ec2Service extends BaseService implements IEc2Service {
       throw new Error('Maximum percent must be 100 for daemon mode.');
     }
 
-    if (props.daemon && props.minHealthyPercent !== undefined && props.minHealthyPercent !== 0) {
-      throw new Error('Minimum healthy percent must be 0 for daemon mode.');
+    if (props.daemon && props.minHealthyPercent !== undefined && props.maxHealthyPercent !== undefined &&
+      props.minHealthyPercent >= props.maxHealthyPercent) {
+      throw new Error('Minimum healthy percent must be less than maximum healthy percent for daemon mode.');
+    }
+
+    if (props.daemon && props.minHealthyPercent !== undefined && props.minHealthyPercent < 0) {
```

Comment:
Do we need to validate this, or will cfn validate it for us (Feels like overkill)? If we're gonna check this logic here, we should be doing it for maxHealthyPercent as well.

Reply:
we have validation for max actually since max should always be 100 when daemon strategy is set